userApi/src/test/java/cs4337/group9/userapi/Service/UserServiceTest.java

```diff
@@ -0,0 +1,198 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private AdminActionRepository adminActionRepository;
+
+    @InjectMocks
+    private UserService userService;
+
+    private UserEntity user;
+    private UUID userId;
+    private UUID adminId;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        userId = UUID.randomUUID();
+        adminId = UUID.randomUUID();
+        user = new UserEntity(userId, "testUser", "test@example.com", "password", Role.USER, Status.ACTIVE, "Test about", "profile.jpg", null, null);
+    }
+
+    @Test
+    void testCreateUser_UserAlreadyExists_ThrowsException() {
+        // Arrange
+        when(userRepository.findByEmail(user.getEmail())).thenReturn(Optional.of(user));
+
+        // Act & Assert
+        UserAlreadyExistsException exception = assertThrows(UserAlreadyExistsException.class, () -> {
+            userService.createUser(user);
+        });
+        assertEquals("User already exists with email: test@example.com", exception.getMessage());
+    }
+
+    @Test
+    void testCreateUser_Success() {
+        // Arrange
+        when(userRepository.findByEmail(user.getEmail())).thenReturn(Optional.empty());
+        when(userRepository.save(any(UserEntity.class))).thenReturn(user);
+
+        // Act
+        userService.createUser(user);
+
+        // Assert
+        verify(userRepository, times(1)).save(user);
+    }
+
+    @Test
+    void testGetUserById_UserNotFound_ThrowsException() {
+        // Arrange
+        when(userRepository.findById(userId)).thenReturn(Optional.empty());
+
+        // Act & Assert
+        UserNotFoundException exception = assertThrows(UserNotFoundException.class, () -> {
+            userService.getUserById(userId);
+        });
+        assertEquals("User not found with id: " + userId, exception.getMessage());
+    }
+
+    @Test
+    void testGetUserById_Success() {
+        // Arrange
+        when(userRepository.findById(userId)).thenReturn(Optional.of(user));
+
+        // Act
+        UserEntity result = userService.getUserById(userId);
+
+        // Assert
+        assertNotNull(result);
+        assertEquals(user.getId(), result.getId());
+    }
+
+    @Test
+    void testUpdateUser_UserNotFound_ThrowsException() {
+        // Arrange
+        UserEntity updatedUser = new UserEntity(userId, "updatedUser", "updated@example.com", "newpassword", Role.USER, Status.ACTIVE, "Updated about", "updated.jpg", null, null);
+        when(userRepository.findById(userId)).thenReturn(Optional.empty());
+
+        // Act & Assert
+        UserNotFoundException exception = assertThrows(UserNotFoundException.class, () -> {
+            userService.updateUser(userId, updatedUser);
+        });
+        assertEquals("User not found with id: " + userId, exception.getMessage());
```

Comment:
The `assertEquals()` message doesnt match the message acutally thrown by the exception 
`Expected :User not found with id: f2e706ad-3deb-4de7-b976-fb103dd6ea04
Actual   :User with ID f2e706ad-3deb-4de7-b976-fb103dd6ea04 not found`